### PR TITLE
James 1704 1Enable MPT test

### PR DESCRIPTION
**Pull request by chibenwa**
In this PR I correct in Memory Mailbox implementation : 
 - Conform to standard unit tests
 - Correct a bug upon SimpleMailboxMessage copy

The corresponding change sets corrected MPT unit tests.

As we plan to rely on in memory for TDD testing, I activated inmemory profile for build.

Also, elasticsearch MPT tests were unactivated. I activated them.

### CODE REVIEW

**Comment by chibenwa** [comment]
Rebased on Antoine's PR for enabling MPT tests, and not merging blindly.... https://github.com/linagora/james-project/pull/186

Here are the fixuped comment written before rebase : 
 - https://github.com/chibenwa/james-project/commit/e4c967d5edd9cc7417556fb4a72f5aee326ff6f3#commitcomment-16637896
 - https://github.com/chibenwa/james-project/commit/9d448ebf79291693b0339b4239b66c79644caf39#commitcomment-16637929

**Comment by aduprat** [comment]
Can one of the admins verify this patch?

**Comment by aduprat** [comment]
Can one of the admins verify this patch?

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
test this please

File: mailbox/memory/src/main/java/org/apache/james/mailbox/inmemory/mail/InMemoryMailboxMapper.java [hunk 1]
```diff
@@ -115,17 +117,21 @@ public void endRequest() {
...
         for (Mailbox<InMemoryId> box:mailboxesById.values()) {
-            if (box.getName().startsWith(mailboxName)) {
+            if (belongsToSameUser(mailbox, box) && box.getName().startsWith(mailboxName)) {
                 return true;
             }
         }
         return false;
     }
 
+    private boolean belongsToSameUser(Mailbox<InMemoryId> mailbox, Mailbox<InMemoryId> box) {
```
Comment (mbaechler) [inline mailbox/memory/src/main/java/org/apache/james/mailbox/inmemory/mail/InMemoryMailboxMapper.java]: badly named arguments

**Comment by aduprat** [comment]
:+1:

**Comment by mbaechler** [comment]
:+1:

**Comment by mbaechler** [comment]
test this please

**Pull request closed without merging**